addons/web_tour_recorder/static/tests/tour_recorder.test.js

```diff
@@ -0,0 +1,396 @@
+    await _checkSteps(["1. .o_button", "2. .o_link"]);
+});
+
+test("Click on element with path that can be reduced", async () => {
+    await mountWithCleanup(`
+        <div class=".o_parent">
+            <div name="field_name">
+                <div class="o_div_2">
+                    <div class="o_div_3 click"></div>
+                </div>
+            </div>
+            <div name="field_partner_id">
+                <div class="o_div_2">
+                    <div class="o_div_3"></div>
+                </div>
+            </div>
+        </div>
+    `);
+
+    expect(".o_tour_recorder").toHaveCount(1);
+    click(".o_button_record");
+    await animationFrame();
+    click(".click");
+    await animationFrame();
+    await _checkSteps(["1. div[name='field_name'] .o_div_3"]);
+});
+
+test("Click on input", async () => {
+    await mountWithCleanup(`
+        <div class=".o_parent">
+            <input type="text" class="click"/>
+        </div>
+    `);
+
+    expect(".o_tour_recorder").toHaveCount(1);
+    click(".o_button_record");
+    await animationFrame();
+    click(".click");
+    await animationFrame();
+    expect(".o_button_record").toHaveText("Record (recording keyboard)");
+    await _checkSteps(["1. input"]);
+});
+
+test("Click on tag that is inside a contenteditable", async () => {
+    await mountWithCleanup(`
+        <div class=".o_parent">
+            <div class="o_editor" contenteditable="true">
+                <p class="click oe-hint oe-command-temporary-hint" placeholder="My placeholder..."></p>
+            </div>
+        </div>
+    `);
+
+    expect(".o_tour_recorder").toHaveCount(1);
+    click(".o_button_record");
+    await animationFrame();
+    click(".click");
+    await animationFrame();
+    expect(".o_button_record").toHaveText("Record (recording keyboard)");
+    await _checkSteps(["1. .o_editor[contenteditable='true']"]);
+});
+
+test("Remove step during recording", async () => {
+    await mountWithCleanup(`
+        <div class="o_parent">
+            <div class="o_child click"></div>
+        </div>
+    `);
+
+    expect(".o_tour_recorder").toHaveCount(1);
+    click(".o_button_record");
+    await animationFrame();
+    click(".click");
+    await animationFrame();
+    await _checkSteps(["1. .o_child"]);
+    click(".o_button_steps");
+    await animationFrame();
+    contains(".o_button_delete_step").click();
+    click(".o_button_steps");
+    await animationFrame();
+    await _checkSteps([]);
+});
+
+test("Edit input", async () => {
+    await mountWithCleanup(`
+        <div class="o_parent">
+            <input type="text" class="click"/>
+        </div>
+    `);
+
+    expect(".o_tour_recorder").toHaveCount(1);
+    click(".o_button_record");
+    await animationFrame();
+    click(".click");
+    await animationFrame();
+    expect(".o_button_record").toHaveText("Record (recording keyboard)");
+    edit("Bismillah");
+    await _checkSteps(["1. input\n(run: edit Bismillah)"]);
```

Comment:
it's a little weird that we're checking the steps purely through inspecting what they render, would definitely be more robust to inspect the internal state of the service (or component if it stays in the component). If for some reason we remove the numbers from the list, or do it in css, or reorganize what's rendered, all these tests will break even though it all still works.